packages/net-stubbing/test/unit/route-matching-spec.ts

```diff
@@ -209,15 +209,53 @@ describe('intercept-request', function () {
         proxiedUrl: 'http://bar.baz/foo?_',
       }
 
-      let prevRoute: BackendRoute
-      let e: string[] = []
+      const e: string[] = []
 
       // @ts-ignore
-      while ((prevRoute = getRouteForRequest(routes, req, prevRoute))) {
-        e.push(prevRoute.id)
+      for (const route of getRoutesForRequest(routes, req)) {
+        e.push(route.id)
       }
 
       expect(e).to.deep.eq(['1', '3', '4', '2'])
     })
+
+    it('yields identical matches', function () {
+      // This is a reproduction of issue #22693
+      const routes: Partial<BackendRoute>[] = [
+        {
+          id: '1',
+          routeMatcher: {
+            pathname: '/foo',
+          },
+        },
+        {
+          id: '1',
+          routeMatcher: {
+            pathname: '/foo',
+          },
+        },
```

Comment:
Should `routes[0]` and `routes[1]` be the same object (by reference) for this to be a true repro of #22693?

Reply:
Having two identical objects was sufficient to cause problems due to how `_.findIndex` is implemented. The test runner would lock up with the original implementation in place. 
I'm still not certain whether the issue we've seen is caused by identical objects or a duplicate registration of the same object...so this repro is a bit speculative.